scipy/spatial/transform/rotation.py

```diff
@@ -1465,6 +1466,42 @@ def magnitude(self):
         else:
             return angles
 
+    def mean(self, return_variance=False):
+        """Get the mean of the rotation(s).
+
+        Returns
+        -------
+        mean : `Rotation` instance
+            Object containing the mean of the rotations in the current
+            instance.
+        return_variance : bool, optional
+            Whether to return the variance of the rotation(s)
+
+        Notes
+        -----
+        The mean used is the chordal L2 mean (also called the projected or
+        induced arithmetic mean).
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> r = R.random(10, np.random.RandomState(0))
+        >>> r.mean().as_quat()
+        array([0.96505487, 0.14713643, 0.17081499, 0.13357476])
+        """
+
+        quat = self._quat.reshape((len(self), 4))
+        K = np.dot(quat.T, quat)
+        l, v = np.linalg.eig(K)
+        index = np.argmax(l)
+        mean = self.__class__(v[:, index], normalized=True)
+
+        if return_variance:
+            variance = np.sum((self.as_dcm() - mean.as_dcm())**2 / len(self))
```

Comment:
I'm not convinced by that computation.

1. Why is the variance of dcm elements is chosen as the measure of variance?
2. At least it should be explained in the docstring.

What if we choose the RMS of rotation magnitudes from the mean rotation to given rotations instead? At least it has the same units as rotation.

Reply:
I see that here you basically computes the objective function.

It needs some more considerations.